### ISSUE TITLE
Why be frugal with MC_KEY?

### ISSUE THREAD

**Issue by LPardue** [GrumpyOldTroll/draft-jholland-quic-multicast#93]
Section 10.2 says:

> A server SHOULD NOT send MC_KEY frames for channels except those the client has joined or will be imminently asked to join.

That sounds sensible but my question is, why shouldn't they? If they ignore the recommendation, what can go wrong?

**Comment by GrumpyOldTroll** [comment]
Good point, thanks.  I think the answer is "principle of least trust" from RFC 4949, which in this instance tells us we should not deliberately leak keys to clients that don't need them, as that would increase the key exposure footprint unnecessarily.
